Destination/RotatableStreamDestination.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Zenstruck\BackupBundle\Destination;
+
+use Psr\Log\LoggerInterface;
+use Zenstruck\BackupBundle\Rotator\Rotator;
+
+class RotatableStreamDestination extends StreamDestination
+{
+    /**
+     * @var null|Rotator
+     */
+    private $preRotator;
+
+    /**
+     * @var null|Rotator
+     */
+    private $postRotator;
+
+    /**
+     * A constructor
+     *
+     * @param string $directory Path to backup directory.
+     * @param Rotator|null $preRotator Rotator to execute prior to push.
+     * @param Rotator|null $postRotator Rotator to execute after push.
+     */
+    public function __construct($directory, Rotator $preRotator = null, Rotator $postRotator = null)
+    {
+        parent::__construct($directory);
+        $this->preRotator = $preRotator;
+        $this->postRotator = $postRotator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function push($filename, LoggerInterface $logger)
```

Comment:
I have to disagree on that, I do not see the reason why there should be a separate interface.

Think of Destination as a bag of stuffs. You put stuffs in there. Is that end of the story?

- Would you like to peak into bag into one point of time?
- Would you like to remove old, rotten stuff from bag?
- Would you like to know how many thing you have in the bag?
- Would you like to take something from bag and give that to someone?

What is a bag if only thing that you can do with it is to put stuff in it?

Ok, if user should not be allowed to remove backup - that is fine, no "remove" method, but Iteration is quite natural thing to have.

I have stated a lot of arguments that supports the necessity of iterating trough backups in one destination. I would like to hear counter-arguments and then I can understand your position.